### DISCUSSION TITLE
Remove requirement to save Read queries

### DISCUSSION THREAD

**Discussion by TheFunkyBiskit** [Budibase/budibase#10440] [Ideas]
Please remove the requirement that a Read query must return values before being saved. I don't see a reason for this except to prevent people from adding queries that don't work, which just seems strange. If I have a query that I know works but don't have data in the database to pull from, or just otherwise don't feel the need to prove it to Budibase that the query works, I should be able to simply save it. Is there another purpose I'm missing here?

**Comment by TheFunkyBiskit** [comment]
Thinking a little deeper on this, is it required so that the resultant schema of the query is known?

**Reply by melohagan** [reply]
That's exactly right! Until you run the Read query Budibase doesn't know what structure your query response will be coming back in.

It also gives you the opportunity to set the data types, remove fields you don't want to include, and transform the response into a suitable format.

**Reply by TheFunkyBiskit** [reply]
Understood, thank you! 😊
